notebooks/Testgrid_flakiness_detection.ipynb

```diff
@@ -0,0 +1,1458 @@
+{
```

Comment:
<p>Line 18: Is 12 or 0 equivalent to a continued failure? I would think 0 (not run) would not continue a failed test count.</p><p> </p><p>Line 27: Why do you ignore more than 3 consecutive failures? ( I know its something we discussed, but we should document it here as it looks like an arbitrary decision in the code) </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aicoe-aiops/ocp-ci-analysis/pull/37/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

Reply:
<p><strong>Regarding line 18</strong>: We are not counting 0 (not run) as a failure but ignoring it. That's why I added line 19.</p><p>Consider a test run [12, 12, 0, 12, 12, 0, 12, 12]</p><p>In the above run, we don't know the value 0 since those test cases are not run. Hence, I am ignoring those runs.</p><p>Converting it to [12, 12, 12, 12, 12, 12]</p><p> </p><p>From my little understanding of test grid dataset.</p><p>We have lots of test cases with results as</p><p> </p><p>[12, 12, 12, 0, 12, 0, 12, 0, 12, 12, 12, 12, 12, 12, 0, 12, 12, 12, 12, 0, 0, 0, 12, 12, 12.......]</p><p>Its more likely that in the above test runs, 0(not runs) would have also been 12(fails).</p><p>Hence, we can not consider 0 as edge or take them into the calculation. Best option is to ignore them.</p><p> </p><p><strong>Regarding line 27:&nbsp;</strong>Done, added comment.</p><p> </p><p> </p>

Reply:
**Regarding 18**: OK, if the test is not run then it should not count towards the flakyness calculation. That makes sense.

But what about the a case like  [12,12,0,0,0,12,1,1], in which you have a break in testing, would you want to consider the first three 12's together as a signal of a legitimate failure? 

Either way, this is fine as the naive, approach, as I think this issues gets addressed with the next method below anyways. 

**Regarding line 27** Where is the updated comment? Is this it? "We ignored more than three consecutive failures and test cases that are not run while calculating the flakiness score" it still doesn't answer the question why we have landed on three instead of 2 or 4 consecutive failures? Does it have to do with the probability of 3 failures in a row, or because we are starting by using the existing testgrid implementation where the same logic is used?

Reply:
<p>Actually, I added the comment on line 28. </p><p>Logic is given on this conversation: <a href="https://github.com/GoogleCloudPlatform/testgrid/pull/224#discussion_r475835903" target="_blank">https://github.com/GoogleCloudPlatform/testgrid/pull/224#discussion_r475835903</a></p><p>Should, I change the comment?</p><p>It is based on current testgrid implementation. </p>